### PR TITLE
Pick up latest markdown language service and clean up server package

### PR DESCRIPTION
**Pull request by mjbvz**
- Picks up newest markdown language service
- Excludes some files from server publish
- Delete unused file
- Update readme
- Align server version with LS version

### CODE REVIEW

**Review by rebornix — APPROVED** [review]

File: extensions/markdown-language-features/server/package/README.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+## Server capabilities
+
+- [Completions](https://microsoft.github.io/language-server-protocol/specification#textDocument_completion) for Markdown links.
+
+- [Folding](https://microsoft.github.io/language-server-protocol/specification#textDocument_foldingRange) of Markdown regions, block elements, and header sections.
+
+- [Smart selection](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_selectionRange) for inline elements, block elements, and header sections.
+
+- [Document Symbols](https://microsoft.github.io/language-server-protocol/specification#textDocument_documentSymbol) for quick navigation to headers in a document.
```
Comment (rzhao271) [inline extensions/markdown-language-features/server/package/README.md:18]: Nit: casing here is inconsistent with above

File: extensions/markdown-language-features/server/package/README.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+#### `markdown/findMarkdownFilesInWorkspace`
+
+Get a list of all markdown files in the workspace.
+
+
+## Contribute
+
+The source code of the Markdown language server can be found in the [VSCode repository](https://github.com/microsoft/vscode) at [extensions/markdown-language-features/server](https://github.com/microsoft/vscode/tree/master/extensions/markdown-language-features/server).
+
+File issues and pull requests in the [VSCode GitHub Issues](https://github.com/microsoft/vscode/issues). See the document [How to Contribute](https://github.com/microsoft/vscode/wiki/How-to-Contribute) on how to build and run from source.
```
Comment (rzhao271) [inline extensions/markdown-language-features/server/package/README.md:117]: Nit: "VSCode GitHub issues tracker" or remove "the" in front of the link text

File: extensions/markdown-language-features/server/package/README.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
+# Markdown Language Server
+
+> **❗ Import** This is still in development. While the language server is being used by VS Code, it has not yet been tested with other clients.
```
Comment (rzhao271) [inline extensions/markdown-language-features/server/package/README.md:3]: Nit: Important instead of import

**Review by rzhao271 — APPROVED** [review]
Nitpicks on the README but LGTM in general

**Pull request merged**